Readability.js

```diff
@@ -721,11 +721,11 @@ Readability.prototype = {
           }
         }
 
-        // Remove empty DIV, SECTION, and HEADER nodes
+        // Remove DIV, SECTION, and HEADER nodes without any content(e.g. text, image, video, or iframe).
```

Comment:
"e.g." means "for example". I'm not clear on what this code is doing and why it's necessary to add a specialcase for elements that contain no images, videos or iframes and no text (but do contain other children). It feels like what we're really doing is hacking around the problem of nested empty things, so if I have:

```
<div>
<div>
</div>
</div>
```

we might be removing the inner div but not the outer one, or something. It feels like we should just explicitly check for that case (so when we remove a node, check if the parent then has no more children and no text using the existing `_isEmptyElement` function).

Alternatively, it seems we should use the same list the if condition here is using to check whether there are any child elements that *aren't* in this set of nodes about which we don't care.

As a counterexample where your code would make a different decision, it looks like this patch will remove containers that contain `<br>` or `<hr>` elements (which wouldn't affect their parent's textContent), which seems wrong.